tests/scripts/test_publish_typescript_package.py

```
@@ -94,15 +94,36 @@ def _exact_view_responses(
         (["view", PACKAGE, f"dist-tags.{dist_tag}"], _result(dist_tag_version)),
+        (["view", package_version, "readme"], _result(readme)),
     ]
 
 
+def test_invalid_readme_encoding_fails_closed(package_directory: Path):
+    (package_directory / "README.md").write_bytes(b"invalid utf-8: \xff")
+    runner = NpmRunner([])
+
+    with pytest.raises(
+        publish_typescript_package.PublicationError,
+        match="Package README.md could not be read",
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)